### ISSUE TITLE
Unsafe multiple inheritance should respect the mro algorithm 

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#433]
```py
class A:
    def __init__(self, a: int):
        super().__init__()
class B:
    pass
class C(B, A):
    pass
```
This reports an error, but as far as I can tell it's safe
